### PR TITLE
add getClientConfiguration accessor to AmazonWebServiceClient

### PR DESCRIPTION
**Pull request by dbyron0**
I'm looking for a way (without reflection) to see what configuration I actually ended up with, so I can log it.

### CODE REVIEW

File: aws-java-sdk-core/src/main/java/com/amazonaws/AmazonWebServiceClient.java [hunk 1]
```diff
@@ -961,4 +961,8 @@ protected boolean calculateCRC32FromCompressedData() {
     public String getSignerOverride() {
         return clientConfiguration.getSignerOverride();
     }
+
+    public ClientConfiguration getClientConfiguration() {
+       return clientConfiguration;
```
Comment (dagnir) [inline aws-java-sdk-core/src/main/java/com/amazonaws/AmazonWebServiceClient.java]: Can we return a copy here?

Comment (dbyron0) [reply aws-java-sdk-core/src/main/java/com/amazonaws/AmazonWebServiceClient.java]: Sure: https://github.com/aws/aws-sdk-java/pull/1641/commits/e85d0dc28024aa7c4ace47cc5beb382d8e452faf

**Comment by zoewangg** [comment]
Thanks for the PR! LGTM. Could you squash your commits into one commit and confirm that this pull request can be released under the Apache 2 license?

**Comment by dbyron0** [comment]
Squashed and rebased.  And yes, the Apache 2 license is fine.

**Comment by zoewangg** [comment]
Thanks for your contribution! The changes will be included in the next release.

**Comment by dbyron0** [comment]
Thanks for merging it!

**Pull request merged**